src/validation.cpp

```diff
@@ -3722,7 +3722,7 @@ bool ChainstateManager::AcceptBlockHeader(const CBlockHeader& block, BlockValida
                         m_blockman.m_dirty_blockindex.insert(invalid_walk);
                         invalid_walk = invalid_walk->pprev;
                     }
-                    LogPrint(BCLog::VALIDATION, "%s: %s prev block invalid\n", __func__, hash.ToString());
+                    LogPrint(BCLog::VALIDATION, "%s has prev block invalid: %s\n", hash.ToString(), block.hashPrevBlock.ToString());
```

Comment:
```suggestion
                    LogPrint(BCLog::VALIDATION, "header %s has prev block invalid: %s\n", hash.ToString(), block.hashPrevBlock.ToString());
```

Sorry for continuing to bike-shed this, but maybe instead of just printing a full hash, clarify that this is a hash of a header?

Reply:
+1. I'd also suggest spelling out "previous" in full, don't see the need to abbreviate. Finally, grammar could be improved a bit. All together, what about:
`"header %s has invalid previous block: %s\n"`

Reply:
There could be a value in keeping the string `"prev block invalid"` the same, to allow `git grep "prev block invalid"` to continue to work, no?

Reply:
Good point, hadn't considered that. Your suggestion seems optimal then.

Reply:
updated to add header clarification